client/client.go

```diff
@@ -829,25 +839,36 @@ func (r *NotaryRepository) validateRoot(rootJSON []byte) (*data.SignedRoot, erro
 // creates and adds one new key or delegates managing the key to the server.
 // These changes are staged in a changelist until publish is called.
 func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
-	if role == data.CanonicalRootRole || role == data.CanonicalTimestampRole {
-		return fmt.Errorf(
-			"notary does not currently support rotating the %s key", role)
-	}
-	if serverManagesKey && role == data.CanonicalTargetsRole {
-		return ErrInvalidRemoteRole{Role: data.CanonicalTargetsRole}
+	switch {
+	// We currently support locally managing targets keys, remotely managing
+	// timestamp keys, and locally or remotely managing snapshot keys.
+	case role == data.CanonicalTargetsRole:
+		if serverManagesKey {
+			return ErrInvalidRemoteRole{Role: data.CanonicalTargetsRole}
+		}
+	case role == data.CanonicalTimestampRole:
+		if !serverManagesKey {
+			return ErrInvalidLocalRole{Role: data.CanonicalTimestampRole}
+		}
+	case role == data.CanonicalSnapshotRole:
```

Comment:
nit: Could we move the snapshot case to the top of the switch with a break to make it more clear that the snapshot is a valid case?